src/einsteinpy/coordinates/utils.py

```diff
@@ -0,0 +1,178 @@
+import numpy as np
+
+from ..ijit import jit
+
+
+def cartesian_to_spherical_fast(
+    x, y, z, v_x=None, v_y=None, v_z=None, velocities_provided=False
+):
+    if velocities_provided:
+        return cartesian_to_spherical(x, y, z, v_x, v_y, v_z)
+    return cartesian_to_spherical_novel(x, y, z)
+
+
+@jit
+def cartesian_to_spherical(x, y, z, v_x, v_y, v_z):
+    """
+    Utility function (jitted) to convert cartesian to spherical.
+    This function should eventually result in Coordinate Transformation Graph!
+    """
+    r = np.sqrt(x ** 2 + y ** 2 + z ** 2)
+    theta = np.arccos(z / r)
+    phi = np.arctan2(y, x)
```

Comment:
you could have done `r, theta, phi = cartesian_to_spherical_novel(x, y, z)`. Otherwise, there is code duplication.

Reply:
I am not sure which versions of numba people might have.